identity/aziot-dps-client-async/src/lib.rs

```diff
@@ -331,11 +335,15 @@ impl Client {
                 let reg_result: model::TpmRegistrationResult = serde_json::from_slice(&body)
                     .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?;
 
-                // update auth method
-                *auth_kind = DpsAuthKind::TpmWithAuth {
-                    auth_key: reg_result.authentication_key,
-                };
+                self.tpm_client
```

Comment:
Maybe a comment here would remind readers (slow readers like me) about the difference between this `import_auth_key` (which is technically importing the DPS nonce and not the primary key for IoT Hub) and the other one outside.

Reply:
And another suggestion for future maintainers/reviewers: to add extra emphasis on the use of an auth key, I would consider renaming `aziot_identity_common::Credentials::Tpm` to `aziot_identity_common::Credentials::TpmAuthKey`

Reply:
We already had a comment in this file (at the top of the `request` method itself), so I just moved it to a more appropriate place + tweaked the wording a bit. I just linked back to the relevant docs about how the DPS TPM flow works.

Let me know if you want to change and wording.